### ISSUE TITLE
[BUG] Invalid hook calls using the documentation suggested approach for handling data relationships in material-ui datagrid 

### ISSUE THREAD

**Issue by gerardosabetta** [pankod/refine#2062]
**Describe the bug**
As of today the docs describe this as the proper way to fetch data from a related field.

```
            {
                field: "category.id",
                headerName: "Category",
                type: "number",
                minWidth: 250,
                flex: 1,
                valueGetter: (params) => {
                    const { data } = useOne<ICategory>({
                        resource: "categories",
                        id: params.row.category.id,
                    });
                    return data?.data.title;
                },
            },
```

This produces the following 3 issues:
1. React complains about the ilegal invocation: 
```
react-dom.development.js:14906 Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```
2. Keyboard controls of the data grid stop working after clicking / focusing the cell with the ilegal invocation.
3. ESLint rule `react-hooks/rules-of-hooks` nags you for using hooks inside of a function.

**To Reproduce**
Try to handle relationships as advised in the docs

**Expected behavior**
Should work without eslint warnings or console errors

**Additional context**

Defering the `useOne` invocation to a real component solves the issue. I used a generic HOC to get rid of this issue.

Here is how I did it in my project. Pardon my inline typescripts hacks. Some of the types were not exported in the library.

![image](https://user-images.githubusercontent.com/55319644/176010409-ce0692ab-ad4f-4e53-8a27-3c55b3b13ec8.png)

**Comment by salihozdemir** [comment]
Hey @gerardosabetta,
Thank you for contacting us. In the coming days, we considered removing the column dependency of the useDataGrid hook to solve this problem. With the new release, we think it will be better to use handle relationships like the following:

```tsx
import React from "react";
import { useMany } from "@pankod/refine-core";
import {
  useDataGrid,
  DataGrid,
  GridColumns,
  List,
  EditButton,
} from "@pankod/refine-mui";

import { ICategory, IPost } from "interfaces";

export const PostList: React.FC = () => {
  const { dataGridProps } = useDataGrid<IPost>();

  const categoryIds = dataGridProps.rows.map((item) => item.category.id);
  const { data: categoriesData, isLoading } = useMany<ICategory>({
    resource: "categories",
    ids: categoryIds,
    queryOptions: {
      enabled: categoryIds.length > 0,
    },
  });

  const columns = React.useMemo<GridColumns<IPost>>(
    () => [
      {
        field: "id",
        headerName: "ID",
        type: "number",
      },
      {
        field: "title",
        headerName: "Title",
      },
      {
        field: "category.id",
        headerName: "Category",
        type: "number",
        renderCell: ({ row }) => {
          if (isLoading) {
            return "Loading...";
          }

          const category = categoriesData?.data.find(
            (item) => item.id === row.category.id
          );
          return category?.title;
        },
      },
      {
        field: "status",
        headerName: "Status",
        minWidth: 120,
        flex: 0.3,
      },
      {
        field: "actions",
        type: "actions",
        headerName: "Actions",
        renderCell: function render({ row }) {
          return <EditButton hideText recordItemId={row.id} />;
        },
        align: "center",
        headerAlign: "center",
        minWidth: 80,
      },
    ],
    [categoriesData, isLoading]
  );

  return (
    <List>
      <DataGrid {...dataGridProps} columns={columns} autoHeight />
    </List>
  );
};

```
It also prevents sending too many requests. 

You can follow the PR that will be opened in the coming days here.

**Comment by omeraplak** [comment]
Hey @gerardosabetta ,
We've released `@pankod/refine-mui@3.37.0` 👊

Now, `columns` property is deprecated. So, If you are getting warning like below:

![image](https://user-images.githubusercontent.com/41580619/177142764-5a83aff3-05c2-44e9-b1cb-ebcc31e77973.png)

You can run the following command to fix it:
```bash
npx @pankod/refine-codemod use-data-grid-columns
```
Or you can remove `columns` property on `useDataGrid`, then give it directly to the `<DataGrid>` component.

**Comment by gerardosabetta** [comment]
You even made a codemod for this, awesome. Refine <3
